### PR TITLE
[ANCHOR-536] validate `SelfDepositInfoGenerator` in SEP6/24/31 Config

### PR DESCRIPTION
**Pull request by JiahuiWho**
### Description

Make distribution account optional

### Context

`distribution_account` is not a required field in asset, and this is validated by AssetsConfigTest and AssetsServiceTest that using test_assets.yaml, which includes no-distribution-account asset.

Also confirmed with @Ifropc with above to add tests for DepositInfoGenerator

### Testing

- `./gradlew test`

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java [hunk 1]
```diff
@@ -212,11 +212,13 @@ public InteractiveTransactionResponse withdraw(
             .sep10Account(token.getAccount())
             .sep10AccountMemo(token.getAccountMemo())
             .fromAccount(sourceAccount)
-            // TODO - jamie to add unique address generator
-            .withdrawAnchorAccount(asset.getDistributionAccount())
             .toAccount(asset.getDistributionAccount())
             .clientDomain(token.getClientDomain());
 
+    if (asset.getDistributionAccount() != null) {
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java:218]: It is safer to use `isEmpty()`

File: core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java [hunk 2]
```diff
@@ -217,12 +217,16 @@ public Sep31PostTransactionResponse postTransaction(
...
             .amountOut(null)
             .amountOutAsset(null)
-            // updateDepositInfo will update these ⬇️
             .stellarAccountId(assetInfo.getDistributionAccount())
             .stellarMemo(null)
             .stellarMemoType(null)
             .build();
 
+    // updateDepositInfo will update these ⬇️
+    if (assetInfo.getDistributionAccount() != null) {
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java:226]: nit: `isEmpty()`

**Comment by lijamie98** [comment]
Please also add the validation of `self` deposit info generator for `SEP-31` and `SEP-6`.

If any of the asset does not have the `distribution_account` configured, the `self` deposit info generator should not be used.

Please also update the `anchor-config-default-values.yaml` file.

File: core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java (hunk 1 above)
Comment (Ifropc) [reply core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java:218]: Wouldn't it throw npe?

Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java:218]: `StringHelper.isEmpty()` won't throw npe.

File: platform/src/main/java/org/stellar/anchor/platform/component/observer/PaymentObserverBeans.java [hunk 3]
```diff
@@ -35,7 +37,10 @@ public StellarPaymentObserver stellarPaymentObserver(
     if (assetService == null || assetService.listAllAssets() == null) {
       throw new ServerErrorException("Asset service cannot be empty.");
     }
-    List<AssetInfo> stellarAssets = assetService.listStellarAssets();
+    List<AssetInfo> stellarAssets =
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/component/observer/PaymentObserverBeans.java]: Isn't it equivalent to `assetService.listStellarAssets`?

**Review by Ifropc — APPROVED** [review]

**Review by lijamie98 — APPROVED** [review]
LGTM.

**Pull request merged**